### DISCUSSION TITLE
NoahMP 4.0.1 is not able run with generic metforcing while NoahMP 3.6 runs for the same generic metforcing.

### DISCUSSION THREAD

**Discussion by PriDeHydro** [NASA-LIS/LISF#1617] [Q&A]
Hi. 

I am trying to run NoahMP4.0.1 with Hymap2 at 0.1 degrees over the portion in southern India using a meteorological forecast generated by the National Center for Medium-Range Weather Forecast (NCMRWF).
NCMRWF forecast is 1 to 3 days hourly forecast at 4km spatial resolution. The forecast is made to resemble LDT Generated forcing with the following attributes
![metldt](https://github.com/user-attachments/assets/bff4ef1b-833b-4e05-a7c2-1f0c99073397)

This LDT runs fine with NoahMP 3.6 but does not run with NoahMP 4.0.1. The error in NoahMP 4.0.1 is shown below
![err](https://github.com/user-attachments/assets/05f2ffca-e923-42d3-9119-86ca8890db89)

_***All runs are made are coldstart runs._

I am attaching a Google Drive link containing the meteorological forecast used for the simulation, the corresponding LDT, and the LIS config file for both NoahMP 3.6 and NoahMP 4.0.1. 

File link: https://drive.google.com/drive/folders/1ii8uSKkiP8NHA-Q9vs8Bxz9p5FSTQi03?usp=sharing

I would appreciate any help regarding this @dmocko, @karsenau,  @sujayvkumar 

Thanks.

**Comment by dmocko** [comment]
Hi @PriDeHydro 

Just to clarify - you mean to say that LIS (and not LDT) appears to be running fine with Noah-MP-3.6 LSM, but not 4.0.1?

Can you attach (or put in the Google drive) all of the lislog files for both LSMs?

I did notice that you have ":Decompose by processes:  .true." in your 4.0.1 config, but not your 3.6 config.
If you remove this line from the 4.0.1 config and try to run again, do you get the same error?

The reason you are getting this error, though, is because there is a grid point that does not have wind forcing data.

Are you sure that the LANDMASK is identical between the two lis_input files?

**Reply by PriDeHydro** [reply]
HI @dmocko 

Thanks for the reply. 

Yes, LIS is running fine with Noah-MP-3.6 LSM, but not 4.0.1.

Attached the lislog files in the google drive. (Link: https://drive.google.com/drive/folders/1ii8uSKkiP8NHA-Q9vs8Bxz9p5FSTQi03?usp=sharing)

Checked with ".false." as well. still, the problem persists.

Yes, I can understand the error. The grid point where the wind forcing is not found is within the domain of both the land mask of NoahMP 3.6 and NoahMP 4.0.1. However, if there is an issue in the forcing, NoahMP 3.6 should have encountered a similar problem, which it did not. 

Land masks are different for both NoahMP 3.6 and NoahMP 4.0.1. But the forcing domain is larger than the land mask for both.

**Reply by PriDeHydro** [reply]
Hi @dmocko 

Did you get a chance to look into the problem? Would really appreciate any help..

Also tagging @karsenau @sujayvkumar

**Reply by dmocko** [reply]
Hi @PriDeHydro 

Why are your lis.config files for the two versions of Noah-MP so different?   Can you clean them both up and make them
more similar, and try them both again?  It's very difficult for me to compare them to try to figure out the issue.

That said, are you sure that Noah-MP-3.6 is actually running?  This line is not correct in your lis_noahmp_36 config file:
`Land surface model:                 "NoahMP.3.6"`
It should be:
`Land surface model:                 "Noah-MP.3.6"`

Please refer to the LIS Users' Guide to ensure you have the correct options as you clean up the config files:
https://nasa-lis.github.io/LISF/LIS_users_guide/LIS_users_guide.html

Furthermore, why are the land masks different between the two versions?  Can you also please share your ldt.config files,
as well as your lis_input files?

**Reply by dmocko** [reply]
Hi @PriDeHydro 

Just following up on this.  Are you now able to run both versions?  They should have the same landmask,
so they should both run with the same forcing.

If not, please share your ldt.config and lis_input files for both LSMs, as well as your corrected lis.config files.

**Reply by PriDeHydro** [reply]
Hi @dmocko 

The problem still persists. 

1. "NoahMP.3.6" works fine while LIS stops if we use"Noah-MP.3.6"
2. While shifting from NoahMP 3.6 to NoahMP 4.0.1 we increased the land mask to cover South Asia.

I have attached the corrected LIS config files, lis log files, and LIS input files in the drive link. The LDT config files for both LSMs are not available at the moment.

**Reply by dmocko** [reply]
HI @PriDeHydro 

You are correct; you should use "NoahMP.3.6".  We need to update our documentation.

You either need to generate new/updated forcing that will cover your new land mask,
or follow one of the options in this post/answer:

https://github.com/NASA-LIS/LISF/discussions/1251
